intel_extension_for_transformers/llm/runtime/graph/models/model_utils/model_utils.h

```diff
@@ -255,6 +255,33 @@ MODEL_API void model_reset_timings(struct model_context* ctx);
 // Print system information
 MODEL_API const char* model_print_system_info(void);
 
+/* kv cache utils */
+// kv cache both stores permuted tensor
+// k shape is [head_dim, N, n_head]
+// v shape is [N, head_dim, n_head] or [N, n_embd]
+/* kv cache utils */
+
+// copy consecutive tokens from one seq to another
+MODEL_API void ne_model_kv_cache_seq_cpy(struct model_context* ctx, const model_seq_id& seq_id_src,
+                                         const model_seq_id& seq_id_dst, const model_pos& p0, const model_pos& p1);
+// MODEL_API void jblas_model_kv_cache_seq_cpy(struct model_context* ctx, const model_seq_id& seq_id_src,
+//                                             const model_seq_id& seq_id_dst, const model_pos& p0, const model_pos&
+//                                             p1);
+MODEL_API void model_kv_cache_seq_cpy(struct model_context* ctx, const model_seq_id& seq_id_src,
+                                      const model_seq_id& seq_id_dst, const model_pos& p0, const model_pos& p1);
+
+// concat several seqs into a continuous batch from kv cache
+MODEL_API ne_tensor* ne_model_kv_cache_seq_concat(struct ne_cgraph* cgraph, struct model_context* moctx,
+                                                  struct ne_context* nectx, const int64_t& ne0, const int64_t& ne1,
+                                                  const int64_t& ne2, const int64_t& ne3,
+                                                  const std::vector<int>& block_ids, const int& layer_idx,
+                                                  const bool& concat_k);
```

Comment:
Remove the declearations of `ne_model_kv_xxx` and make them static because they are helper functions to be used only inside `model_utils.cpp`

Reply:
fixed in [8ba8390](https://github.com/intel/intel-extension-for-transformers/pull/583/commits/8ba83901db1fc87880050fae145e83c160258631)